addon/-private/system/relationships/state/belongs-to.js

```diff
@@ -84,6 +84,7 @@ export default class BelongsToRelationship extends Relationship {
     //temporary fix to not remove newly created records if server returned null.
     //TODO remove once we have proper diffing
     if (this.inverseModelData && this.inverseModelData.isNew() && !this.canonicalState) {
+      this.willSync = false;
```

Comment:
The `willSync` flag is maintained in the super `Relationship` class and it will be reset when `flushCanonical` is called. However, in this special case, where the relationship is new and the server don't return any value for the relationship, the flag will remain as `willSync = true` and any subsequent attempts to `flushCanonical` the relationship will be ignored.